### DISCUSSION TITLE
vite+ts: can't use "this" in getters, how to solve?

### DISCUSSION THREAD

**Discussion by llcat** [vuejs/pinia#1502] [Help and Questions] [answered]
reproduct url:  https://stackblitz.com/edit/vitejs-vite-guherm?file=package.json

![image](https://user-images.githubusercontent.com/17900822/182108374-af518e35-4397-46e6-9dda-0aaab18be0dd.png)

it's not pinia issues, i find some related issue, but not with pinia
https://github.com/vitejs/vite/discussions/8640
https://github.com/evanw/esbuild/issues/2328
https://github.com/vitejs/vite/issues/8644

this warning can slient by follow config
![image](https://user-images.githubusercontent.com/17900822/182111287-395fd40c-42d8-4719-bd95-03f232da09d2.png)

help wanted

**Comment by llcat** [comment, marked as answer]
it seems this warning is caused by mine error use way, i should not use arrow function, but use normal function still have type error
![image](https://user-images.githubusercontent.com/17900822/182277955-8883211f-098e-431c-9dfc-c1fbe2bc91e7.png)

**Reply by posva** [reply]
add the return type to the getter (https://pinia.vuejs.org/core-concepts/getters.html#accessing-other-getters)

**Reply by posva** [reply]
you probably want to learn more about how `this` (the context) works in JS and arrow functions too! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Comment by llcat** [comment]
😸 thanks, add return type work well, i really should learn `this` context again, love pinia and your great job!

**Reply by llcat** [reply]
as a ts beginer, make a summary, always add return type in getters 📖 
```ts
/** Store augmented with getters. For internal usage only. For internal use **only** */
export declare type _StoreWithGetters<G> = {
    readonly [k in keyof G]: G[k] extends (...args: any[]) => infer R ? R : UnwrapRef<G[k]>;
};
```
